apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/ecosystem-header.client.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+import { Spinner } from "@/components/ui/Spinner/Spinner";
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { CheckIcon, ChevronsUpDown } from "lucide-react";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { useMemo } from "react";
+import invariant from "tiny-invariant";
+import { useEcosystemList } from "../../../hooks/use-ecosystem-list";
+
+export function EcosystemHeader({ ecosystemId }: { ecosystemId: string }) {
+  const { ecosystems, isFetched } = useEcosystemList();
+  const router = useRouter();
+  const currentEcosystem = useMemo(
+    () => ecosystems?.find((ecosystem) => ecosystem.id === ecosystemId),
+    [ecosystems, ecosystemId],
+  );
+
+  if (
+    isFetched &&
+    !ecosystems?.find((ecosystem) => ecosystem.id === ecosystemId)
+  ) {
+    // No ecosystem found, redirect to 404
+    router.replace("/404");
```

Comment:
potentially instead redirect back to the "ecosystem home" page? feels more actionable for the user?

Reply:
also feels odd to redirect in the "header" I would expect this logic in the "page" itself?

Reply:
Originally, I had it redirecting to the ecosystem home page but it felt confusing to the user if they had a link to a dead ecosystem or something. They get no feedback that the ecosystem no longer exists (maybe we could put up a toast, not sure if that would persist between client-side routes)

Reply:
idea:
- put an error toast (which will persist across navigation)
- then immediately redirect to the home page

Reply:
Sounds good. Just remembered why I put it in the header. This level is going to have multiple pages (only one right now is permissions) with the tabs layout, and I don't want to repeat this check on every page. This should really go in the /[ecosystemId] layout but I want to keep that server-side to easily add metadata to it.

Reply:
I _think_ put it in the layout for now, ultimately we're going to (have to) fix the auth cookie situation at which point all the logic that is in the client here today will move to the server anyways